src/modules/shared/pages/innovation/documents/document-newdit.config.ts

```diff
@@ -96,54 +111,100 @@ export const WIZARD_WITH_LOCATION_QUESTIONS: WizardEngineModel = new WizardEngin
+  runtimeRules: [(steps: WizardStepType[], currentValues: StepPayloadType, currentStep: number | 'summary') => wizardWithLocationRuntimeRules(steps, currentValues, currentStep)],
   inboundParsing: (data: InboundPayloadType) => inboundParsing({ ...data, wizardType: 'WIZARD_WITH_LOCATION_QUESTIONS' }),
   outboundParsing: (data: StepPayloadType) => outboundParsing(data),
   summaryParsing: (data: StepPayloadType) => summaryParsing(data)
 });
 
 
-function runtimeRules(steps: WizardStepType[], currentValues: StepPayloadType, currentStep: number | 'summary'): void {
+function wizardWithLocationRuntimeRules(steps: WizardStepType[], data: StepPayloadType, currentStep: number | 'summary'): void {
+
+  // As we need to do async calls, we do it only one time at the begginning.
+  // Response will be received after this method finishes, but that's not a problem as this information is only needed on step 3.
+  if (!data.evidencesList) {
+    innovationStoreInnovationService.getSectionInfo(data.innovationId, evidencesSectionId, {}).subscribe(response => {
+      if (response.data.hasEvidence === 'YES') { // Only show evidences if section 2.2 question is YES.
+        data.evidencesList = ((response.data.evidences as StepPayloadType['evidencesList']) ?? []).map(item => ({ id: item.id, name: item.name, summary: StringsHelper.smartTruncate(item.summary, 150) }));
+      } else {
+        data.evidencesList = [];
+      }
+    });
+  }
+
 
   steps.splice(1);
 
-  if (currentValues.contextType === 'INNOVATION_SECTION') {
+  if (data.relatedWithSection === 'YES') {
 
     steps.push(new FormEngineModel({
       parameters: [{
         id: 'section', dataType: 'radio-group', label: stepsLabels.l2.label,
         validations: { isRequired: [true, 'Choose one option'] },
-        items: innovationSectionsItems
+        items: innovationSectionsItems.map(item => ({
+          ...item,
+          ...(item.value === evidencesSectionId && (data.evidencesList ?? []).length > 0 && {
+            description: `There's ${data.evidencesList?.length} evidences. Choosing this section we'll ask you more information on the next question.`
+          })
+        }))
       }]
     }));
 
+    if (data.section === evidencesSectionId && (data.evidencesList ?? []).length > 0) {
+
+      steps.push(new FormEngineModel({
+        parameters: [{
+          id: 'evidence', dataType: 'radio-group', label: stepsLabels.l3.label,
+          validations: { isRequired: [true, 'Choose one option'] },
+          items: [
+            ...[{ value: 'NONE', label: `I want to upload this document on section "${innovationSectionsItems.find(item => item.value === evidencesSectionId)?.label}", not an evidence` }],
+            ...(data.evidencesList ?? []).map(item => ({ value: item.id, label: item.name, description: item.summary })) // TODO: limit this summary!
```

Comment:
I think this comment can be removed since you are already doing it when you the service call

Reply:
Removed! ;)